### ISSUE TITLE
linter: rule options are dropped from `--print-config` when using object-form `extends` in `oxlint.config.ts`

### ISSUE THREAD

**Issue by wakamsha** [oxc-project/oxc#22230]
### What version of Oxlint are you using?

1.63.0

### What command did you run?

`oxlint -c .oxlint.config.ts --print-config`

#### `rules/eslint.ts`

```ts
import { defineConfig } from 'oxlint';

export default defineConfig({
  categories: { correctness: 'error' /* ... */ },
  rules: {
    'getter-return': ['error', { allowImplicit: true }],
    'no-console': ['warn', { allow: ['info', 'warn', 'error', 'time', 'timeEnd'] }],
    'no-implicit-coercion': ['error', { boolean: false, number: true, string: true, allow: [] }],
    'object-shorthand': ['error', 'always', { avoidQuotes: true }],
    'prefer-const': ['error', { destructuring: 'any', ignoreReadBeforeAssign: true }],
    // ...
  },
});
```

#### `rules/oxc.ts`

```ts
import { defineConfig } from 'oxlint';

export default defineConfig({
  categories: { correctness: 'error' /* ... */ },
});
```

#### `oxlint.config.ts`

```ts
import { defineConfig } from 'oxlint';
import eslint from './rules/eslint.ts';
import oxc from './rules/oxc.ts';

export default defineConfig({
  plugins: ['eslint', 'oxc'],
  extends: [eslint, oxc],
});
```

### What happened?

When `--print-config` is used with a config that pulls in another config object via `extends`, the rule **severities** from the extended configs are correctly merged into the printed output, but the **per-rule options** (the `[severity, ...options]` tail) are silently dropped from every rule.

#### Reproduction:

1. Use the two files above.
2. Run `pnpm oxlint -c ./rules/eslint.ts --print-config > eslint.json`.
3. Run `pnpm oxlint -c ./oxlint.config.ts --print-config > config.json`.
4. Diff the two outputs against rules that declare options.

#### Expected:

The printed config from `oxlint.config.ts` retains the option arrays from `eslint.ts`, e.g.:

```jsonc
{
  "rules": {
    "getter-return": ["deny", [{ "allowImplicit": true }]],
    "no-console": ["warn", [{ "allow": ["info", "warn", "error", "time", "timeEnd"] }]],
    "object-shorthand": ["deny", ["always", { "avoidQuotes": true }]]
    // ...
  }
}
```

#### Actual:

Severities are merged correctly, but options are stripped:

```jsonc
{
  "rules": {
    "getter-return": "deny",
    "no-console": "warn",
    "object-shorthand": "deny"
    // ...
  }
}
```

When `eslint.ts` is used directly as the root config (`-c ./rules/eslint.ts`), the same rules are printed with their option arrays intact, so the loss only occurs when the rules originate from an object inside `extends`.

Importantly, **lint behavior is unaffected** — running the linter against a fixture confirms that the options are honored at lint time. The bug is confined to the `--print-config` output path.

#### Root cause:

`ConfigStoreBuilder::resolve_final_config_file` in `crates/oxc_linter/src/config/config_builder.rs` builds the printed `rules` map by zipping two sources:

```rust
pub fn resolve_final_config_file(&self, oxlintrc: Oxlintrc) -> String {
    let mut oxlintrc = oxlintrc;
    let previous_rules = std::mem::take(&mut oxlintrc.rules);

    let rule_name_to_rule = previous_rules.rules.into_iter()
        .map(|r| (get_name(&r.plugin_name, &r.rule_name), r))
        .collect::<rustc_hash::FxHashMap<_, _>>();

    let new_rules = self.rules.iter()
        .map(|(r, severity)| ESLintRule {
            severity: *severity,
            config: rule_name_to_rule
                .get(&get_name(r.plugin_name(), r.name()))
                .map(|r| r.config.clone())
                .unwrap_or_default(),
            // ...
        })
```

The severities come from `self.rules` (the merged builder state, populated via `OxlintRules::override_rules`), but the option arrays are looked up from `oxlintrc.rules` — and the `oxlintrc` passed in from `apps/oxlint/src/lint.rs:303` is the **un-merged** `root_config`:

```rust
return crate::mode::run_print_config(&config_builder, root_config, stdout);
```

In the object-form `extends` case, `root_config.rules` only contains the rules defined directly on the top-level config (often empty, as in `oxlint.config.ts`). Rules inherited through `extends_configs` live on the merged config that `from_oxlintrc` produces internally and discards. As a result, the lookup table is empty for those rules and `unwrap_or_default()` substitutes an empty options array.

The merge itself preserves options correctly (`Oxlintrc::merge` in `oxlintrc.rs` clones whole `ESLintRule` structs), and `from_configuration` bakes them into the resolved `RuleEnum` instances used at lint time — that is why runtime linting works while `--print-config` does not.

This is closely related to #17724, but distinct: #17724 is about path-form `extends` in JSON configs not being expanded at all. Here, object-form `extends` *is* expanded for severities; the regression is specific to options being stripped.

#### Possible directions:

- Pass the merged `Oxlintrc` (the one `from_oxlintrc` already produces internally via `resolve_oxlintrc_config`) to `run_print_config` instead of `root_config`, so the options lookup table includes inherited rules.
- Alternatively, walk `extends_configs` recursively when building `rule_name_to_rule` inside `resolve_final_config_file`, mirroring the merge order so that closer extends win.

Thank you.
